tests/generic_config_updater/test_acl.py

```diff
@@ -0,0 +1,360 @@
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_res_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname, cfg_facts):
+    """
+    Setup/teardown fixture for acl config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+        cfg_facts: config facts for selected DUT
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    config_tmpfile = generate_tmpfile(duthost)
+    logger.info("config_tmpfile {} Backing up config_db.json".format(config_tmpfile))
+    duthost.shell("sudo cp /etc/sonic/config_db.json {}".format(config_tmpfile))
+
+    # Cleanup acl config
+    duthost.shell('sonic-db-cli CONFIG_DB keys "ACL_RULE|*" | xargs sonic-db-cli CONFIG_DB del',
+        module_ignore_errors=True)
+    duthost.shell('sonic-db-cli CONFIG_DB keys "ACL_TABLE|*" | xargs sonic-db-cli CONFIG_DB del',
+        module_ignore_errors=True)
+
+    yield
+
+    logger.info("Restoring config_db.json")
+    duthost.shell("sudo cp {} /etc/sonic/config_db.json".format(config_tmpfile))
+    delete_tmpfile(duthost, config_tmpfile)
+
+     # Cleanup acl config
+    config_reload(duthost)
+
+def expect_res_success_acl_table(duthost, expected_content_list, unexpected_content_list):
+    """Check if acl table show as expected
+    """
+    cmds = "show acl table"
+    output = duthost.shell(cmds)
+    pytest_assert(not output['rc'], "'{}' is not running successfully".format(cmds))
+
+    expect_res_success(duthost, output, expected_content_list, unexpected_content_list)
+
+def expect_res_success_acl_rule(duthost, expected_content_list, unexpected_content_list):
+    """Check if acl rule added as expected
+    """
+    cmds = "sudo iptables -S"
```

Comment:
>-S [](http://example.com/codeflow?start=26&length=2)

What is the reason to use `-S` instead of `-nL` ?

Reply:
Refering to [test_cacl_application.py #L436](https://github.com/Azure/sonic-mgmt/blob/master/tests/cacl/test_cacl_application.py#L436)